spring-integration-redis/src/test/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandlerTests.java

```diff
@@ -0,0 +1,177 @@
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.integration.handler.ReactiveMessageHandlerAdapter;
+import org.springframework.integration.redis.rules.RedisAvailable;
+import org.springframework.integration.redis.rules.RedisAvailableTests;
+import org.springframework.integration.redis.util.Address;
+import org.springframework.integration.redis.util.Person;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * @author Attoumane Ahamadi
+ *
+ * @since 5.4
+ */
+@ContextConfiguration(classes = ReactiveRedisStreamMessageHandlerTestsContext.class)
+@RunWith(SpringRunner.class)
+@DirtiesContext
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ReactiveRedisStreamMessageHandlerTests extends RedisAvailableTests {
+
+	@Autowired
+	@Qualifier("streamChannel")
+	private MessageChannel messageChannel;
+
+	@Autowired
+	private ReactiveRedisConnectionFactory redisConnectionFactory;
+
+	@Autowired
+	private ReactiveMessageHandlerAdapter handlerAdapter;
+
+	@Autowired
+	private ReactiveRedisStreamMessageHandler streamMessageHandler;
+
+	@Test
+	@RedisAvailable
+	public void integrationStreamOutboundTest() {
+		String messagePayload = "Hello stream message";
+
+		messageChannel.send(new GenericMessage<>(messagePayload));
+
+		RedisSerializationContext<String, String> serializationContext = redisStringOrJsonSerializationContext(null);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, serializationContext);
+
+		ObjectRecord<String, String> record = template.opsForStream().read(String.class, StreamOffset.fromStart(ReactiveRedisStreamMessageHandlerTestsContext.STREAM_KEY)).blockFirst();
+
+		assertThat(record.getStream()).isEqualTo(ReactiveRedisStreamMessageHandlerTestsContext.STREAM_KEY);
+		assertThat(record.getValue()).isEqualTo(messagePayload);
+
+		template.delete(ReactiveRedisStreamMessageHandlerTestsContext.STREAM_KEY).block();
+	}
+
+	//TODO Find why the deserialization fail does not work
+	/*@Test
+	@RedisAvailable*/
+	public void explicitJsonSerializationContextTest() {
+		List<String> messagePayload = Arrays.asList("Hello", "stream", "message");
+
+		RedisSerializationContext<String, Object> jsonSerializationContext = redisStringOrJsonSerializationContext(List.class);
+
+		streamMessageHandler.setSerializationContext(jsonSerializationContext);
```

Comment:
You set a fresh context, but you don't recreate a template.
You need to perform this before sending a message:
```
streamMessageHandler.afterPropertiesSet();
```

Reply:
Is it not set by reference?

Reply:
Huh?
No, according your code it is done like this:
```
public void setSerializationContext(RedisSerializationContext serializationContext) {
		Assert.notNull(serializationContext, "'serializationContext' must not be null");
		this.serializationContext = serializationContext;
}
```
But we definitely need to populate it into the `ReactiveRedisTemplate` which you create exactly in the `onInit()`:
```
ReactiveRedisTemplate template = new ReactiveRedisTemplate<>(this.connectionFactory,
				this.serializationContext);
```
So, changing the `serializationContext` on the `ReactiveRedisStreamMessageHandler` should lead to the `afterPropertiesSet()` call.
Technically I don't see problem with this, but we should follow our own API. There I suggest respective fix for your tests.